### PR TITLE
Lesson 17

### PR DESCRIPTION
**Pull request by Elena-Nam**
found dog API, styled dog page

### CODE REVIEW

**Comment by Elena-Nam** [comment]
Hello, Lee. I am still having an issue to filter drinks. Also I need an advice how to create reservation form with chosen dog and drink.

File: js/index.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+const url = "https://api.github.com/repos/Elena-Nam/Open-API-Project";
+fetch(url)
+  .then(response => response.json())
+  .then(data => console.log(data))
+  .catch((e) => console.log(e));
+
+
+  document.getElementById('year').textContent = new Date().getFullYear();
```
Comment (Lee-Rice) [inline js/index.js:9]: I'm not sure if you need this file. I don't see where it's currently being used.

File: dogs.html [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            <div id="popup" class="popup" style="display: none;">
+                <div class="popup-content">
+                  <span class="close" onclick="closePopup()">&times;</span>
+                  <h3 id="popup-title">Dog Information</h3>
+                  <p><strong>Age:</strong> <span id="popup-age"></span></p>
+                  <p><strong>Sex:</strong> <span id="popup-sex"></span></p>
+                  <p><strong>Size:</strong> <span id="popup-size"></span></p>
+                  <p><strong>Location:</strong> <span id="popup-location"></span></p>
+                </div>
+              </div>
```
Comment (Lee-Rice) [inline dogs.html:48]: Good thinking allowing the user to get additional information about a dog before scheduling to meet them.

File: dogs.html [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        
+        </div> <!-- closing div for container-main -->
+
+        <div class = "container_two_buttons">
+            <button class="back"><a href="index.html">&laquo; Back </a></button>
+            <button class="next"><a href="reservation.html">Next &raquo;</a></button>
+            </div>
+            <footer>
+                <a href = "https://www.adoptapet.com" target="_blank"><img src = "adoptapet.png" alt = "logo of adoptapet" width= 200px;></a>
+            </footer>
```
Comment (Lee-Rice) [inline dogs.html:59]: I like how this footer acts as a placeholder until the information about the dogs has loaded.

File: drinks.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                 <a class="active"><strong> Coffee </strong></a> 
+                 <a href="dogs.html"><strong> Dogs </strong></a> 
+                 <a href="reservation.html"><strong> Reservation </strong></a>
+                 </nav>
+            </div>
+        </header>
+        <section id ="coffee">
+            <h2> Choose a drink </h2>
+            <div id="myBtnContainer">
+                <button class="btn active" onclick="filterSelection('all')">Show all  </button>
```
Comment (Lee-Rice) [inline drinks.html:32]: I think only cold drinks are currently being shown for the `Show all button`.

File: js/dogs.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
+
+const url = 'https://api-staging.adoptapet.com/search/pet_search?key=hg4nsv85lppeoqqixy3tnlt3k8lj6o0c&v=3&output=json&city_or_zip=29445&geo_range=50&species=dog&breed_id=real=801&sex=m&age=young&color_id=54&pet_size_range_id=2&hair=&bonded_pair=&special_needs=&include_mixes=&added_after=&start_number=1&end_number=50&meta_only=0';
```
Comment (Lee-Rice) [inline js/dogs.js:2]: Just as a best practices comment: When you're using an API you never want to store your API key in a public repo. Instead, you'd want to store them securely using something like [GitHub Secrets](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions) or [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).

**Review by Lee-Rice — COMMENTED** [review]
This is a huge change and must have been a ton of work.  It looks great. Just a couple suggestions:
- It would be best to put all your images in a single `images` folder.
- Be sure to double-check your project against the project requirements.

As far as how to transfer selections between functions, this is the point where I would spin up a database to store all the data you're going to need. Since databases haven't been covered yet, I'm going to suggest using [sessionStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage). In a real world application a database would work better since you'll need to store a lot of information for each user and have it persist between browser sessions, but I think this method will be fine for a demonstration project.

For your question on how to separate the other drinks from the hot coffee list, you could check the ingredients of the returned drink and exclude all those that don't contain either coffee, espresso, etc.

Please let me know if you have any questions on this. Good on you for going above and beyond on this project. This is impressive. Keep up the good work.

File: js/index.js (hunk 1 above)
Comment (Elena-Nam) [reply js/index.js:9]: It was lesson-15 assignment "... Repeat the portfolio part of the assignment above for this project.." (https://classes.codethedream.org/course/intro-to-programming-v4/io?week=15&lesson=Fetch+API) when we began API project

**Pull request merged**